src/Spectre.libHeatmapDataScaling/HistogramEqualization.cpp

```diff
@@ -0,0 +1,57 @@
+/*
+* HistogramEqualization.cpp
+* Class with histogram equalization algorithm implementation for contrast enhancement.
+*
+Copyright 2018 Daniel Babiak
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "Spectre.libFunctional/Transform.h"
+#include "Spectre.libHeatmapDataScaling/HistogramEqualization.h"
+
+namespace spectre::visualization
+{
+std::vector<double> HistogramEqualization::scaleData(const gsl::span<const double> intensities)
+{
+    std::vector<double> roundedIntensities = spectre::core::functional::transform(intensities, [](double intensity) { return static_cast<int>(intensity + 0.5); });
```

Comment:
Is it neccessary to keep it as a vector of doubles if we round it to integer values? Maybe some `unsigned` or `unsigned long` would suffice?